vista/app/src/components/Compartir.js

```diff
@@ -13,14 +13,26 @@ import {sendCompartir} from '../actions/compartirAction';
 */
 class Compartir extends Component {
 
+  constructor (props) {
+    super(props);
+    this.state = {
+      wasShared: props.shared
+    }
+  }
+
   clickHandler = () => {
     let {id, typeContent} = this.props;
+    this.setState({
+      wasShared: !this.state.wasShared
+    });
     this.props.sendCompartir(id, typeContent);
   }
 
   render () {
+    let shared = 'compartirIcon compartido',
+        notShared = 'compartirIcon';
     return (
-        <Avatar className='compartirIcon' icon={'share-alt'} onClick={this.clickHandler}/>
+        <Avatar className={this.state.wasShared ? shared : notShared} icon={'share-alt'} onClick={this.clickHandler}/>
```

Comment:
Es preferible que directamente se tome el valor de `this.prop.shared` en vez de `this.state.wasShared`. Lo conveniente sería actualizar el estado del elemento en el dataset guardado en el store, de esa manera no tendríamos que hacer la validación y, si la acción asincrónica falla el estado interno del componente se vería afectado y la UI quedaría desactualizada con respecto a la data real. Igualmente, para nuestro caso actual, funciona.